tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationRuntimeTest.java

```diff
@@ -47,153 +46,145 @@
-//
-//        annotValue = fieldAnnotMap.get(StringUtils.fromString("X"));
-//        Assert.assertEquals(TypeChecker.getType(annotValue).getTag(), TypeTags.MAP_TAG);
-//
-//        MapValueImpl<BString, Object> mapValue = (MapValueImpl<BString, Object>) annotValue;
-//        Assert.assertEquals(mapValue.size(), 1);
-//        Assert.assertEquals(mapValue.get(StringUtils.fromString("p")), 2L);
-//
-//        annotatableType = (AnnotatableType) ((TypedescValue) tupleValue.get(1)).getDescribingType();
-//        Assert.assertEquals(annotatableType.getAnnotation(StringUtils.fromString("W")), true);
-//        fieldAnnots = annotatableType.getAnnotation(StringUtils.fromString("$field$.j"));
-//        Assert.assertEquals(TypeChecker.getType(fieldAnnots).getTag(), TypeTags.MAP_TAG);
-//        fieldAnnotMap = (MapValueImpl<BString, Object>) fieldAnnots;
-//
-//        annotValue = fieldAnnotMap.get(StringUtils.fromString("Z"));
-//        Assert.assertEquals(TypeChecker.getType(annotValue).getTag(), TypeTags.BOOLEAN_TAG);
-//        Assert.assertTrue((Boolean) annotValue);
-//
-//        annotValue = fieldAnnotMap.get(StringUtils.fromString("Y"));
-//        Assert.assertEquals(TypeChecker.getType(annotValue).getTag(), TypeTags.MAP_TAG);
-//
-//        mapValue = (MapValueImpl<BString, Object>) annotValue;
-//        Assert.assertEquals(mapValue.size(), 2);
-//        Assert.assertEquals(mapValue.get(StringUtils.fromString("q")).toString(), "hello");
-//        Assert.assertEquals(mapValue.get(StringUtils.fromString("r")).toString(), "world");
-//    }
-//
-//    public void testRecordTypeAnnotationReadonlyValueEdit() {
-//        BRunUtil.invoke(resultAccessNegative, "testRecordTypeAnnotationReadonlyValueEdit");
-//    }
-//
-//    public void testAnnotationOnObjectTypeReadonlyValueEdit() {
-//        BRunUtil.invoke(resultAccessNegative, "testAnnotationOnObjectTypeReadonlyValueEdit");
-//    }
-//
-//    public void testAnnotationOnFunctionTypeReadonlyValueEdit() {
-//        BRunUtil.invoke(resultAccessNegative, "testAnnotationOnFunctionTypeReadonlyValueEdit");
-//    }
-//
-//    public void testReadonlyTypeAnnotationAttachment() {
-//        BRunUtil.invoke(readOnlyValues, "testReadonlyTypeAnnotationAttachment");
-//    }
+    private CompileResult resultOne;
+    private CompileResult resultTwo;
+    private CompileResult resultThree;
+    private CompileResult resultFour;
+    private CompileResult resultAccessNegative;
+    private CompileResult readOnlyValues;
+
+    @BeforeClass
+    public void setup() {
+        resultOne = BCompileUtil.compile("test-src/annotations/annot_access.bal");
+        Assert.assertEquals(resultOne.getErrorCount(), 0);
+
+        resultTwo = BCompileUtil.compile("test-src/annotations/annot_access_with_source_only_points.bal");
+        Assert.assertEquals(resultTwo.getErrorCount(), 0);
+
+        resultThree = BCompileUtil.compile("test-src/annotations/annotations_constant_propagation.bal");
```

Comment:
If the `CompileResult` field is used only within a single method that does not use a data provider, let's move the compilation to that method itself.

Reply:
Done.